bokeh/charts/_data_adapter.py

```diff
@@ -60,7 +63,14 @@ def __init__(self, data, index=None, columns=None, force_alias=True):
                 columns = list(keys())
 
             elif keys is None:
-                columns = list(map(str, range(len(data))))
+                if blaze and isinstance(self._values, blaze.interactive.InteractiveSymbol):
```

Comment:
The `InteractiveSymbol` object is the object that wraps any other kind of data (like a database table).  Interacting with `Expr` more generally allows you to handle queries on that table

```python
>>> d = bz.Data('sqlite:///...')  # InteractiveSymbol object
>>> d[d.name == 'Alice'][[col1, col2, col3]]  # complex query, inherits from Expr
```